packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -57,9 +69,43 @@ function ListViewBlockSelectButton(
 		onDragStart?.( event );
 	};
 
+	/**
+	 * @param {KeyboardEvent} event
+	 */
 	function onKeyDownHandler( event ) {
 		if ( event.keyCode === ENTER || event.keyCode === SPACE ) {
 			onClick( event );
+		} else if (
+			event.keyCode === BACKSPACE ||
+			event.keyCode === DELETE ||
+			isMatch( 'core/block-editor/remove', event )
+		) {
+			const selectedBlockClientIds = getSelectedBlockClientIds();
+			const isDeletingSelectedBlocks =
+				selectedBlockClientIds.includes( clientId );
+			const firstBlockClientId = isDeletingSelectedBlocks
+				? selectedBlockClientIds[ 0 ]
+				: clientId;
+
+			let blockToSelect =
+				getPreviousBlockClientId( firstBlockClientId ) ??
+				// If the previous block is not found (when the first block is deleted),
+				// fallback to focus the parent block.
+				getBlockRootClientId( firstBlockClientId );
+
+			removeBlocks(
+				isDeletingSelectedBlocks
+					? selectedBlockClientIds
+					: [ clientId ],
+				false
+			);
+
+			// If there's no previous block nor parent block, focus the first block.
+			if ( ! blockToSelect ) {
+				blockToSelect = getBlockOrder()[ 0 ];
+			}
+
+			updateSelection( blockToSelect );
```

Comment:
Should the `updateSelection` only fire if `isDeletingSelectedBlocks` is `truthy`?

I noticed a subtle issue that if I delete a block outside of the selection, after deletion, the selection updates to the block directly above the currently focused one:

![2023-05-11 13 58 39](https://github.com/WordPress/gutenberg/assets/14988353/43539e9f-4752-4c5c-a244-99c8a90cc70d)